src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,211 @@
+            ResultSet resultSet = getStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car by id: "
+                    + id + " from DB", e);
+        }
+        if (car != null) {
+            car.setDrivers(getAllDriversByCarId(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllRequest = "SELECT * FROM cars WHERE is_deleted = false";
+        List<Car> allCars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllStatement = connection.createStatement()) {
+            ResultSet resultSet =
+                    getAllStatement.executeQuery(getAllRequest);
+            while (resultSet.next()) {
+                allCars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all cars from DB", e);
+        }
+        allCars.forEach(car -> car.setDrivers(getAllDriversByCarId(car.getId())));
+        return allCars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars SET model = ?, "
+                + "manufacturer_id = ? WHERE id = ? and is_deleted = false;";
+        int updateCount;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement =
+                        connection.prepareStatement(updateRequest)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateCount = updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update car by id: "
+                    + car.getId() + " from DB", e);
+        }
+        if (updateCount > 0) {
+            deleteDrivers(car);
+            insertDrivers(car);
+            return car;
+        }
+        throw new DataProcessingException("Car with id:" + car.getId() + "wasn`t found in DB");
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE cars SET is_deleted = true where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement =
+                        connection.prepareStatement(deleteRequest)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete car by id: "
+                    + id + " from DB", e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car =
```

Comment:
Let's parse manufacturer manually, with the current impl we open nested exception to db, which is bad. Code duplication will be the lesser evil in this case.
So we don't need t ManufacturerService here.

Reply:
changed